src/structlog/processors.py

```diff
@@ -909,3 +911,75 @@ def __call__(
                 event_dict["event"] = replace_by
 
         return event_dict
+
+
+class CallsiteNamespaceAddr:
+    """
+    Attempt to identify and add the caller namespace to the event dict
+    under the ``namespace`` key.
+
+    Arguments:
+
+        levels:
+            A set of log levels to add the ``namespace`` key and
+            information to. An empty set == *
+
+    .. versionadded:: 23.3.0
+    """
+
+    def __init__(self, levels: set[str] | list[str] | None = None):
+        self.levels = levels
+
+    def __call__(
+        self, logger: WrappedLogger, name: str, event_dict: EventDict
+    ) -> EventDict:
+        if self.levels and name not in self.levels:
+            return event_dict
+
+        f, _ = _find_first_app_frame_and_name()
+        event_dict["namespace"] = self.get_qual_name(f)
+
+        return event_dict
+
+    def get_qual_name(self, frame: FrameType) -> str:
```

Comment:
This should not be a (public) method, but a function in `_frames.py`, no?

Reply:
Agreed -- moved